### PR TITLE
Fix/node 12 changes

### PR DESCRIPTION
**Pull request by owenniblock**
## Motivation

We want to be able to build shipwright docker images with tags on node 12. Versioning information is currently broken so this should fix it.

### LTS Logic

I've also fixed some wonky logic on the LTS detection code while I was there.

### Notes

Should we consider pushing this upstream?

Tests all pass locally for me (the LTS bug caused a failing test)

I've just lifted @gordey4doronin's fix from the Issue discussion without testing it 😂 the tests still pass so I assume all is fine but I'll check with Gordey tomorrow before we think about merging this.

### CODE REVIEW

**Comment by owenniblock** [comment]
Build is broken 💥 I'll check it out in the morning - I'm done for now 😄

File: spec/git.spec.js [hunk 1]
```diff
@@ -24,11 +24,13 @@ describe('Git', function () {
       var version = /v([0-9]+)/.exec(process.version)[ 1 ]
       var today = new Date()
       var lts = git.checkLTS()
-      if (version === '6' && today < new Date('10-01-2017')) {
```
Comment (wraithgar) [inline spec/git.spec.js:27]: Let's just do `new Date(2017, 9, 1)` so we aren't fighting localization settings. (reminder that month is 0-indexed on `new Date()`)

**Comment by gordey4doronin** [comment]
Tests are actually all green. The CI fails with this:
```
/home/travis/build/npm-wharf/buildgoggles/node_modules/coveralls/bin/coveralls.js:18

        throw err;

        ^

Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}

The command "nyc npm test && nyc report --reporter=text-lcov | coveralls" exited with 1.
```

**Comment by gordey4doronin** [comment]
So this is an issue with `coveralls` reports. Might be something related to auth token.

File: spec/git.spec.js (hunk 1 above)
Comment (gordey4doronin) [reply spec/git.spec.js:27]: Assuming I'm in UTC+7 there are differences in these three options. I'd keep @owenniblock's choice. 

```js
> new Date('2017-01-10')
2017-01-10T00:00:00.000Z

> new Date('10-01-2017')
2017-09-30T17:00:00.000Z

> new Date(2017, 9, 1)
2017-09-30T17:00:00.000Z
```

**But in general I'd rather prefer to use ISO strings everywhere.**
```js
new Date('2017-01-10T00:00:00.000Z')
```

File: src/git.js [hunk 2]
```diff
@@ -8,15 +8,17 @@ const LATEST = 'latest'
 
 function checkLTS () {
   const version = /v([0-9]+)/.exec(process.version)[ 1 ]
+  const today = new Date()
+  const tomorrow = new Date(today.getTime() + 86400000)
   const schedule = {
-    '6': new Date('2016-10-18'),
-    '8': new Date('2017-10-01'),
-    '10': new Date('2018-10-01')
+    '6': [new Date('2016-10-18'), new Date('2017-10-01')],
```
Comment (gordey4doronin) [inline src/git.js]:
```suggestion
    '6': [new Date('2016-10-18'), new Date('2017-10-31')],
```

File: src/git.js [hunk 3]
```diff
@@ -8,15 +8,17 @@ const LATEST = 'latest'
...
+    '8': [new Date('2017-10-01'), new Date('2018-10-01')],
```
Comment (gordey4doronin) [inline src/git.js]:
```suggestion
    '8': [new Date('2017-10-31'), new Date('2018-10-30')],
```

File: src/git.js [hunk 4]
```diff
@@ -8,15 +8,17 @@ const LATEST = 'latest'
...
+    '10': [new Date('2018-10-01'), new Date('2019-10-21')],
```
Comment (gordey4doronin) [inline src/git.js]:
```suggestion
    '10': [new Date('2018-10-30'), new Date('2019-10-21')],
```

File: src/git.js [hunk 5]
```diff
@@ -8,15 +8,17 @@ const LATEST = 'latest'
...
+    '12': [new Date('2019-10-21'), tomorrow]
   }
-  const startDate = schedule[ version ]
-  if (startDate) {
-    const today = new Date()
-    return today >= startDate
+  const dateRange = schedule[ version ]
```
Comment (gordey4doronin) [inline src/git.js]:
```suggestion
  const dateRange = schedule[version]
```

**Comment by gordey4doronin** [comment]
https://github.com/nodejs/Release

Screenshot 2020-06-10 at 14 23 36

Screenshot 2020-06-10 at 14 23 43

File: package-lock.json [hunk 6]
```diff
@@ -0,0 +1,5437 @@
+{
+  "name": "buildgoggles",
```
Comment (gordey4doronin) [inline package-lock.json]:
```suggestion
  "name": "@npm-wharf/buildgoggles",
```

File: package-lock.json [hunk 7]
```diff
@@ -0,0 +1,5437 @@
...
+  "version": "0.5.2",
```
Comment (gordey4doronin) [inline package-lock.json]:
```suggestion
  "version": "1.0.0",
```

File: src/git.js (hunk 5 above)
Comment (gordey4doronin) [reply src/git.js]: However, I think it's fine to keep existing styles for now. So feel free to ignore this one. 👍

File: spec/git.spec.js (hunk 1 above)
Comment (gordey4doronin) [reply spec/git.spec.js:27]: However, I think ISO stuff is out of scope of this PR. I'm good to keep as is.

**Comment by gordey4doronin** [comment]
I've setup `coveralls` for `npm-wharf/buildgoggles` and re-triggered the builds.

Screenshot 2020-06-10 at 15 14 14

**All is green now.**

**Review by gordey4doronin — APPROVED** [review]
Well played! ✋

**Pull request merged**